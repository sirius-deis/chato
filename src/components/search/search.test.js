import { render, screen, fireEvent } from "@testing-library/react";
import Search from "./search";

describe("Search component", () => {
  it("should match snapshot", () => {
    const fn = jest.fn();
    const { container } = render(<Search onChange={fn}></Search>);
    expect(container).toMatchSnapshot();
  });
  it("should should fire a change event", () => {
    const fn = jest.fn();
    render(<Search onChange={fn}></Search>);
    fireEvent.change(screen.getByPlaceholderText(/Search.../), {
      target: { value: "test" },
    });
    expect(fn).toHaveBeenCalledWith("test");
  });
  it("should clear search input", async () => {
    const fn = jest.fn();
    render(<Search onChange={fn} value="not empty"></Search>);
    fireEvent.click(screen.getByTestId("clear"));
    expect(fn).toHaveBeenCalled();
  });
});
