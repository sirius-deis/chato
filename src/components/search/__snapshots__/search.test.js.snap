// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Search component should match snapshot 1`] = `
<div>
  <div
    class="sc-dcJsrY gMhUjw"
  >
    <input
      class="sc-iGgWBj fgzRSD"
      name="search"
      placeholder="Search..."
      type="search"
    />
  </div>
</div>
`;
